src/ray/gcs/gcs_server/gcs_resource_manager.cc

```diff
@@ -41,29 +44,35 @@ void GcsResourceManager::HandleUpdateResources(
     rpc::SendReplyCallback send_reply_callback) {
   NodeID node_id = NodeID::FromBinary(request.node_id());
   RAY_LOG(DEBUG) << "Updating resources, node id = " << node_id;
-  auto iter = cluster_resources_.find(node_id);
-  std::unordered_map<std::string, double> to_be_updated_resources;
-  for (const auto &entry : request.resources()) {
-    to_be_updated_resources.emplace(entry.first, entry.second.resource_capacity());
+  std::unordered_map<std::string, double> changed_resources;
+  for (auto &entry : request.resources()) {
+    changed_resources.emplace(entry.first, entry.second.resource_capacity());
```

Comment:
Aren't we supposed to overwrite resources in this case? Is emplace overwriting existing keys? This is what I found;

```
Inserts a new element into the container constructed in-place with the given args if there is no element with the key in the container.
```

This seems like we should just use
```cpp
changed_resources[entry.first] = entry.second.resource_capacity;
```
?

Reply:
`request.resources()` is a map, so it can guarantee the uniqueness of the key.